### PR TITLE
[16.0][MIG] hr_timesheet_task_domain: Migration to version 16.0

### PR DESCRIPTION
**Pull request by Yadier-Tecnativa**
@Tecnativa TT43228

supersede for https://github.com/OCA/timesheet/pull/586

### CODE REVIEW

File: hr_timesheet_task_domain/tests/test_hr_timesheet_task_domain.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        self.project = self.env["project.project"].create({"name": "Test project"})
+        self.analytic_account = self.project.analytic_account_id
+        self.task = self.env["project.task"].create(
+            {"name": "Test task", "project_id": self.project.id}
+        )
+        self.line = self.env["account.analytic.line"].create(
+            {
+                "task_id": self.task.id,
+                "account_id": self.analytic_account.id,
+                "name": "Test line",
```
Comment (stefan-tecnativa) [inline hr_timesheet_task_domain/tests/test_hr_timesheet_task_domain.py]:
```suggestion
                "name": "Test line",
                "employee_id": 2,
```

We need an employee when we create a timesheet. This corrects the tests.

Comment (pedrobaeza) [reply hr_timesheet_task_domain/tests/test_hr_timesheet_task_domain.py]: @Yadier-Tecnativa complete this

**Comment by Yadier-Tecnativa** [comment]
Done!

File: hr_timesheet_task_domain/models/account_analytic_line.py [hunk 2]
```diff
@@ -4,28 +4,18 @@
...
-            res_domain = res.setdefault("domain", {})
-            res_domain.update({"task_id": task_domain})
-        else:  # Reset domain for allowing selection of any task
-            res["domain"] = {"task_id": []}
-        if task.project_id == self.project_id:
-            # Restore previous task if belongs to the same project
-            self.task_id = task
-        return res
+    task_id = fields.Many2one(
+        comodel_name="project.task",
```
Comment (pedrobaeza) [inline hr_timesheet_task_domain/models/account_analytic_line.py]: no need to specify again the comodel_name.

File: hr_timesheet_task_domain/README.rst [hunk 3]
```diff
@@ -23,11 +23,10 @@ Task Log: limit Task by Project
     :target: https://runbot.odoo-community.org/runbot/117/15.0
     :alt: Try me on Runbot
 
-|badge1| |badge2| |badge3| |badge4| |badge5| 
+|badge1| |badge2| |badge3| |badge4| |badge5|
```
Comment (pedrobaeza) [inline hr_timesheet_task_domain/README.rst]: Have you edited this file manually? If so, at least keep the trailing space.

**Review by pedrobaeza — COMMENTED** [review]
Add yourself to contributors.

File: hr_timesheet_task_domain/README.rst (hunk 3 above)
Comment (Yadier-Tecnativa) [reply hr_timesheet_task_domain/README.rst]: changed

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**